apps/dashboard/src/components/cursor-pagination.tsx

```diff
@@ -0,0 +1,51 @@
+import { Button } from '@/components/primitives/button';
+import { ChevronLeftIcon, ChevronRightIcon } from '@radix-ui/react-icons';
+
+interface CursorPaginationProps {
+  hasNext: boolean;
+  hasPrevious: boolean;
+  onNext: () => void;
+  onPrevious: () => void;
+  onFirst: () => void;
+}
+
+export function CursorPagination({ hasNext, hasPrevious, onNext, onPrevious, onFirst }: CursorPaginationProps) {
+  return (
+    <div className="bottom-0 mt-auto border-t border-t-neutral-200 bg-white py-3">
+      <div className="flex items-center justify-center px-6">
+        <div className="border-input inline-flex items-center rounded-lg border bg-transparent">
+          <Button
+            variant="secondary"
+            mode="ghost"
+            disabled={!hasPrevious}
+            onClick={onFirst}
+            className="rounded-r-none border-0"
+          >
+            <div className="flex items-center">
+              <ChevronLeftIcon className="size-3" />
+              <ChevronLeftIcon className="-ml-2 size-3" />
+            </div>
+          </Button>
```

Comment:
We also have the button-group component from align which you might be able to use to handle all the radius issues:

```
<ButtonGroup.Root>
        <ButtonGroup.Item>
          <ButtonGroup.Icon as={RiLayoutGridLine} />
          Grid view
        </ButtonGroup.Item>
        <ButtonGroup.Item>
          <ButtonGroup.Icon as={RiListCheck} />
          List view
        </ButtonGroup.Item>
        <ButtonGroup.Item>
          <ButtonGroup.Icon as={RiLayout2Line} />
          Gallery view
        </ButtonGroup.Item>
      </ButtonGroup.Root>
      ```